### ISSUE TITLE
[0.5 Crash] ui_compositor.c or tui.c assertion with indent-blankline & matchit plugins

### ISSUE THREAD

**Issue by bluz71** [neovim/neovim#14267]
Hello Neovim team,

- `nvim --version`: NVIM v0.5.0-dev+1197-ga17782042

- Operating System: Linux Mint 20.1

- Minimal init.vim:
```viml
set mouse=a
set termguicolors

call plug#begin('~/.vim/plugged')
Plug 'lukas-reineke/indent-blankline.nvim', {'branch': 'lua'}
call plug#end()

runtime macros/matchit.vim
```

- Ruby file to edit: [ruby.rb](https://github.com/bluz71/vim-nightfly-guicolors/blob/master/samples/ruby.rb)

- Mouse click to line 28 and do:

- `o` then enter the following text `<CR>def index2<CR><Esc>` (`<CR>` being Enter key)
- `u` to undo

Tip, the quicker the better. Open the Ruby file, click to line 28, do the edit then undo. If it does not crash the first time, exit Neovim and come back in. Maybe I am imagining this, but I feel speed is a factor. Anyway, about 1 in 5 times I am seeing a crash as noted in this screenshot:

![Crash](https://user-images.githubusercontent.com/11382509/113254603-8aa9a480-9312-11eb-8a0c-028e4eab82b0.png)

Note, sometimes I get `ui_compositor.c:565 Assertion` crash, other times I get `tui.c:1390: tui_raw_line: Assertion`.

Best regards.

**Comment by erw7** [comment]
I've tried both manually and with the following steps, but I can't reproduce it successfully.

#### test.vim
```vim
call termopen(["nvim", "-u",  "NORC",  "--cmd",  "set rtp+=/home/erw7/tmp/indent-blankline.nvim", "+set mouse=a | set termguicolors | runtime macros/matchit.vim"])
call chansend(b:terminal_job_id, ":e ruby.rb\<CR>")
redraw
sleep 500m
call chansend(b:terminal_job_id, "\<ESC>[\<lt>0;1;29Modef index2\<CR>\<ESC>")
sleep 500m
redraw
call chansend(b:terminal_job_id, "u")
```
```
> git clone https://github.com/lukas-reineke/indent-blankline.nvim ~/tmp/indent-blankline.nvim
> wget https://raw.githubusercontent.com/bluz71/vim-nightfly-guicolors/master/samples/ruby.rb
> nvim -u NORC
:so test.vim
```

**Comment by bluz71** [comment]
Why the sleeps? In my experience, speed is a factor, the edit and undo needs to happen quickly.

If you can't reproduce, tell me what I need to do to trigger a full stack trace to help you guys debug this since I can reproduce this fairly easily.

**Comment by bluz71** [comment]
Damn, I can't reproduce this at all today.

I could do it so easily yesterday. Sigh.

I think it is best to close this. If I can repeat it I will re-open.

**Comment by erw7** [comment]
> Why the sleeps? In my experience, speed is a factor, the edit and undo needs to happen quickly.

I also `chansend()` everything at once without `sleep` at first. However, I could not reproduce the phenomenon. I guessed that this might happen depending on the timing, so I tried several times, varying the wait time by sleep. Sending input by `chansend()` is processed much faster than input by human hands. I have experienced problems that occur when typing manually, but not when using `chansend()`.
